### PR TITLE
added validator list interface

### PR DESCRIPTION
**Pull request by jmatosp**
Tests for Validators Set 
Transforming ValidatorSet into an interface
Fixed VET issues for core/types
re-generated code for core/types

### CODE REVIEW

File: core/types/validator_test.go [hunk 1]
```diff
@@ -1,3 +1,97 @@
...
+	"testing"
+)
+
+func TestValidator_Properties(t *testing.T) {
+	address := common.Address{}
+	deposit := uint64(100)
+	weight := &big.Int{}
+	validator := NewValidator(address, deposit, weight)
+
+	assert.Equal(t, address, validator.Address())
```
[the remark this answers is not in the document]
Comment (jmatosp) [reply core/types/validator_test.go]: never used `require` doesnt that stop tests from running?

[the remark this answers is not in the document]
Comment (acroca) [reply core/types/validator_test.go]: I think the `require` stops right away and the `assert` continues running.

[the remark this answers is not in the document]
Comment (jmatosp) [reply core/types/validator_test.go]: Ok, will have a try locally, but would prefer to have a full test suit run.maybe require will be useful for test setup

File: core/types/validator.go [hunk 2]
```diff
@@ -27,63 +27,76 @@ func (val *Validator) Address() common.Address { return val.address }
...
-	set.proposer = item.Value.(*Validator)
+	set.proposer = proposer
 
 	// decrement the validator weight since he has been selected
 	set.proposer.weight.Sub(set.proposer.weight, big.NewInt(int64(set.proposer.deposit)))
 }
 
-func (set *ValidatorSet) AtIndex(i int) *Validator {
+func (set *validatorList) At(i int) *Validator {
+	if i < 0 || i > len(set.validators) {
```
Comment (acroca) [inline core/types/validator.go]: actually `i > len(set.validators)` has to be `i >= len(set.validators)` right?

File: core/types/validator_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmatosp) [reply core/types/validator_test.go]: I've checked. It does terminate current test scenario, so it is useful for things like require.NoErr() or requires.NotNil(t, objUnderTest), prob others.

For normal expectations like comparing expected values assert sounds better.

What you think?

File: core/types/validator.go (hunk 2 above)
Comment (jmatosp) [reply core/types/validator.go]: better right a test for it :)

File: core/types/validator_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (acroca) [reply core/types/validator_test.go]: Sure. I generally like it to stop asap when there's an error, but we can use `require` only when steps are required for next calls and `assert` for normal assertions

**Pull request merged**